acme/tests/client_test.py

```diff
@@ -739,11 +739,26 @@ def setUp(self):
             status=messages.STATUS_PENDING,
             authorizations=(self.authzr.uri, self.authzr_uri2),
             finalize='https://www.letsencrypt-demo.org/acme/acct/1/order/1/finalize')
+
         self.orderr = messages.OrderResource(
             body=self.order,
             uri='https://www.letsencrypt-demo.org/acme/acct/1/order/1',
             authorizations=[self.authzr, self.authzr2], csr_pem=CSR_MIXED_PEM)
 
+        self.not_after = datetime.datetime.now(datetime.timezone.utc) + datetime.timedelta(hours=1)
+
+        self.order_with_not_after = messages.Order(
+            identifiers=(self.authz.identifier, self.authz2.identifier),
+            status=messages.STATUS_PENDING,
+            authorizations=(self.authzr.uri, self.authzr_uri2),
+            finalize='https://www.letsencrypt-demo.org/acme/acct/1/order/1/finalize',
+            not_after=self.not_after)
+
+        self.orderr_with_not_after = messages.OrderResource(
+            body=self.order_with_not_after,
+            uri='https://www.letsencrypt-demo.org/acme/acct/1/order/1',
+            authorizations=[self.authzr, self.authzr2], csr_pem=CSR_MIXED_PEM)
+
```

Comment:
nit: if we're not using these in other unit tests, let's fold it all into `test_new_order_with_not_after`.